src/instructions/mod.rs

```diff
@@ -2,11 +2,26 @@ mod add;
 
 pub use add::Add;
 
-use crate::{memory::Memory, registers::Registers};
+use crate::{memory::Memory, opcodes::Opcode, registers::Registers};
 
-pub trait Instruction {
-    fn from_bits(bits: u16) -> Self
-    where
-        Self: Sized;
-    fn execute(&self, registers: &mut Registers, memory: &mut Memory);
+pub enum Instruction {
+    Add(Add),
+}
```

Comment:
It took me a little time to understand what `Add(Add)` exactly is, but I think that if you add documentation later on the project it will be ok.